server/utils/helpers/chat/convertTo.js

```diff
@@ -5,16 +5,18 @@ const { Workspace } = require("../../../models/workspace");
 const { WorkspaceChats } = require("../../../models/workspaceChats");
 
 // Todo: make this more useful for export by adding other columns about workspace, user, time, etc for post-filtering.
-async function convertToCSV(workspaceChatsMap) {
-  const rows = ["role,content"];
-  for (const workspaceChats of Object.values(workspaceChatsMap)) {
-    for (const message of workspaceChats.messages) {
-      // Escape double quotes and wrap content in double quotes
-      const escapedContent = `"${message.content
-        .replace(/"/g, '""')
-        .replace(/\n/g, " ")}"`;
-      rows.push(`${message.role},${escapedContent}`);
-    }
+async function convertToCSV(preparedData) {
+  const rows = ["id,username,workspace,prompt,response,sent_at"];
+  for (const item of preparedData) {
+    // Escape double quotes and wrap content in double quotes
+    const escapedPrompt = `"${item.prompt
+      .replace(/"/g, '""')
+      .replace(/\n/g, " ")}"`;
+    const escapedResponse = `"${item.response
+      .replace(/"/g, '""')
+      .replace(/\n/g, " ")}"`;
+    const row = `"${item.id}","${item.username}","${item.workspace}",${escapedPrompt},${escapedResponse},"${item.sent_at}"`;
+    rows.push(row);
```

Comment:
It would probably be more readable to define a data array that then makes the string
```javascript
const record = [
item.id,
escape(item.username), // we dont validate this to not contain quotes and better safe??
escape(item.workspace), // this needs to be escape because we dont validate this on backend.
escape(item.prompt)
escape(item.response)
item.sent_at
].join(',')

row.push(record)
```